packages/angular/cli/lib/examples/signal-forms/03-cross-field-validation.md

```diff
@@ -0,0 +1,178 @@
+
+## When to Use
+
+Use this pattern when the validity of one form field depends on the value of another. This is common in registration forms, password change forms, and any form where two fields must be consistent with each other. The recommended approach is to apply a validator to the field where the error should be reported (e.g., `confirmPassword`) and use the `valueOf` function in the validation context to reactively get the value of the other field it depends on (e.g., `password`).
+
+## Key Concepts
+
+- **`validate()`:** A function used within a schema to add a validator to a specific field.
+- **`valueOf()`:** A function available in the validation context that allows you to reactively access the value of any other field in the form.
+- **`submit()`:** An async helper function that manages the form's submission state and should be called from the submit event handler.
+
+## Example Files
+
+This example consists of a standalone component that defines and manages a password change form.
+
+### password-form.component.ts
+
+This file defines the component's logic. The schema applies a validator to `confirmPassword` that compares its value to the value of the `password` field.
+
+```typescript
+import { Component, signal, output, ChangeDetectionStrategy } from '@angular/core';
+import {
+  form,
+  schema,
+  validate,
+  requiredError,
+  minLengthError,
+  customError,
+  submit,
+  Control,
+} from '@angular/forms/signals';
+import { JsonPipe } from '@angular/common';
+
+export interface PasswordForm {
+  password: string;
+  confirmPassword: string;
+}
+
+const passwordSchema = schema<PasswordForm>((passwordForm) => {
+  validate(passwordForm.password, ({ value }) => {
+    if (value() === '') return requiredError();
+    if (value().length < 8) return minLengthError(8);
+    return null;
+  });
+
+  validate(passwordForm.confirmPassword, ({ value, valueOf }) => {
+    if (value() !== valueOf(passwordForm.password)) {
+      return customError({ kind: 'mismatch' });
+    }
+    return null;
+  });
+});
+
+@Component({
+  selector: 'app-password-form',
+  imports: [JsonPipe, Control],
+  templateUrl: './password-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PasswordFormComponent {
+  readonly submitted = output<PasswordForm>();
+
+  passwordModel = signal<PasswordForm>({
+    password: '',
+    confirmPassword: '',
+  });
+
+  passwordForm = form(this.passwordModel, passwordSchema);
+
+  async handleSubmit() {
+    await submit(this.passwordForm, async () => {
+      this.submitted.emit(this.passwordForm().value());
+    });
+  }
+}
+```
+
+### password-form.component.html
+
+This file provides the template for the form, showing how to display errors for both field-level and cross-field validation rules.
+
+```html
+<form (submit)="handleSubmit(); $event.preventDefault()">
```

Comment:
For the purpose of the example I think it's best to include the novalidate, because without, native HTML5 form validation will kick in first.

```suggestion
<form (submit)="handleSubmit(); $event.preventDefault()" novalidate>
```